src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -387,6 +409,11 @@ private async Task GetTableChangesAsync(SqlConnection connection, CancellationTo
                 this._rows = new List<IReadOnlyDictionary<string, object>>();
                 this._logger.LogError($"Failed to check for changes in table '{this._userTable.FullName}' due to exception: {e.GetType()}. Exception message: {e.Message}");
                 TelemetryInstance.TrackException(TelemetryErrorName.GetChanges, e, this._telemetryProps);
+                if (e.IsFatalSqlException())
```

Comment:
Why not throw every exception?

Reply:
That's the design for now, we don't want to exit the loop completely and instead retry if an error occurs.

A follow up task is to investigate all possible retry logic for the function - which may change that behavior at that time (most likely implementing some sort of "dead letter queue" like [ServiceBus](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-dead-letter-queues#the-dead-letter-queue) does so that we can extract problematic data and the user can investigate and re-enter it if needed)